src/parser/AstNode.ts

```diff
@@ -47,16 +47,23 @@ export abstract class AstNode {
             if (node.symbolTable) {
                 return node.symbolTable;
             }
-            node = node.parent;
+            //justification: we are following a chain of nodes until we get to one with a SymbolTable,
+            //and the top-level node will always have a SymbolTable
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
In theory I like having it on in most projects, but I think it's getting more noisy as I go on. There's lots of cases where it gives false positives so I think it would be reasonable to disable the rule entirely.